### PR TITLE
Replaced relative link with absolute links and remove locale

### PR DESCRIPTION
**Pull request by sima-zhu**
The PR is to turn on the link check flag.
1. Disable the relative links.
2. Remove locale "en-us" from link.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ With this in mind, there are many tenets or principles that we follow in order t
 
 - Unlike our other language SDKs, many things (such as composing an HTTP pipeline of policies) are done in source code as opposed to runtime. This reduces code size, improves execution speed and locks-in behavior, reducing the chance of bugs at runtime.
 
-- We support microcontrollers with no operating system, microcontrollers with a real-time operating system (like [Azure RTOS](https://azure.microsoft.com/en-us/services/rtos/)), Linux, and Windows. Customers can implement custom platform layers to use our SDK on custom devices.  We provide some platform layers, and encourage the community to submit platform layers to increase the out-of-the-box supported platforms.
+- We support microcontrollers with no operating system, microcontrollers with a real-time operating system (like [Azure RTOS](https://azure.microsoft.com/services/rtos/)), Linux, and Windows. Customers can implement custom platform layers to use our SDK on custom devices.  We provide some platform layers, and encourage the community to submit platform layers to increase the out-of-the-box supported platforms.
```
Comment (ahsonkhan) [inline README.md:19]: Good catch.

File: README.md [hunk 2]
```diff
@@ -454,7 +454,7 @@ See the complete cmake file and how to link your own library [here](https://gith
 
 ## SDK Architecture
 
-At the heart of our SDK is, what we refer to as, [Azure Core](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/docs/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an [Azure Storage Blob](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/docs/storage) client library and [Azure IoT client libraries](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/docs/iot). Here are some of the features that customers use directly:
+At the heart of our SDK is, what we refer to as, [Azure Core](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/docs/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an [Azure Storage Blob](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/docs/storage) client library and [Azure IoT client libraries](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/docs/iot). Here are some of the features that customers use directly:
```
Comment (ahsonkhan) [inline README.md:457]: Question: Why do we prefer `/blob/` over `/tree/`?

File: SECURITY.md [hunk 3]
```diff
@@ -36,6 +36,6 @@ We prefer all communications to be in English.
 
 ## Policy
 
-Microsoft follows the principle of [Coordinated Vulnerability Disclosure](https://www.microsoft.com/en-us/msrc/cvd).
+Microsoft follows the principle of [Coordinated Vulnerability Disclosure](https://www.microsoft.com/msrc/cvd).
```
Comment (ahsonkhan) [inline SECURITY.md:39]: The SECURITY.md content is the verbatim text all new repos are getting seeded with (there's a comment tag at the top and bottom of the file indicating the text version).

For now, please revert the changes to this file.

File: sdk/docs/core/README.md [hunk 4]
```diff
@@ -162,7 +162,7 @@ If you'd like to contribute to this library, please read the [contributing guide
 Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.
 
 <!-- LINKS -->
-[azure_sdk_for_c_contributing]: ../../../CONTRIBUTING.md
+[azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md
 [azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/master/LICENSE
 [azure_sdk_for_c_contributing_developer_guide]: ../../../CONTRIBUTING.md#developer-guide
```
Comment (ahsonkhan) [inline sdk/docs/core/README.md:167]: What about these links?

File: sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md [hunk 5]
```diff
@@ -254,4 +254,4 @@ If you'd like to contribute to this library, please read the [contributing guide
 
 ### License
 
-Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.
\ No newline at end of file
+Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.
```
Comment (ahsonkhan) [inline sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: What's the change here?

**Review by ahsonkhan — COMMENTED** [review]
Revert `SECURITY.md` changes. Otherwise, looks good.

**Review by ahsonkhan — APPROVED** [review]

File: sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md (hunk 5 above)
Comment (momuno) [reply sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: it's a ghost character!

**Comment by momuno** [comment]
awesome, thanks for the cleanup!

**Review by momuno — APPROVED** [review]

File: sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md (hunk 5 above)
Comment (sima-zhu) [reply sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: I am wondering the same

File: README.md (hunk 2 above)
Comment (sima-zhu) [reply README.md:457]: Actually, github link is able to switch each other when necessary. I am using a script to do the replacement, so I just choose one of them as I feel like blob is used more often than tree. Just personal feeling

File: SECURITY.md (hunk 3 above)
Comment (sima-zhu) [reply SECURITY.md:39]: Will do

File: sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md (hunk 5 above)
Comment (ahsonkhan) [reply sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: Is it a line-ending changed from CRLF to LF or vice versa? You can use hexdump to see the exact byte contents of the file and that will show you the difference:
https://www.fileformat.info/tool/hexdump.htm

Comment (momuno) [reply sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: I bet it's a new line at the end of the file.

Comment (momuno) [reply sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: "> \ No newline at end of file
+Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.
"

Was in the email.

Comment (momuno) [reply sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md:257]: except now this is all obsolete because I just merged in a change to this same file... Hopefully it's not much extra work for you!

File: README.md (hunk 2 above)
Comment (weshaggard) [reply README.md:457]: In general blob should be used for files and tree should be used for directories but github usually forwards any blob requests that point to a directory to tree.

File: sdk/docs/core/README.md (hunk 4 above)
Comment (weshaggard) [reply sdk/docs/core/README.md:167]: I suspect these were not flagged because they are not referenced anywhere in the MD file. They should probably be removed.

File: sdk/docs/storage/README.md [hunk 6]
```diff
@@ -81,18 +81,18 @@ Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] li
 <!-- LINKS -->
 [azure_pattern_circuit_breaker]: https://docs.microsoft.com/azure/architecture/patterns/circuit-breaker
 [azure_pattern_retry]: https://docs.microsoft.com/azure/architecture/patterns/retry
-[azure_sdk_for_c_contributing]: ../../../CONTRIBUTING.md
+[azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md
 [azure_sdk_for_c_contributing_developer_guide]: ../../../CONTRIBUTING.md#developer-guide
```
Comment (weshaggard) [inline sdk/docs/storage/README.md:85]: These also look to be unused.

**Review by weshaggard — APPROVED** [review]
A couple of unused links but otherwise looks good.

File: sdk/docs/storage/README.md (hunk 6 above)
Comment (sima-zhu) [reply sdk/docs/storage/README.md:85]: Right, that's why we did not fetch them out. Thanks! I will fix it for now.

File: SECURITY.md (hunk 3 above)
Comment (weshaggard) [reply SECURITY.md]: If we revert this the checks will fail. I think we should go ahead and update this as the link is definitely incorrect.
